### PR TITLE
Added several new bindings

### PR DESCRIPTION
**Pull request by neslib**
- Small bugfix in igSliderFloat2
- Added ImDrawData_ScaleClipRects
- Added ImGuiTextFilter APIs
- Added ImGuiTextEditCallbackData
- Added ImGuiStorate APIs
- Added ImFontAtlas_GetGlyphRanges\* APIs

### CODE REVIEW

**Comment by procedural** [comment]
I guess `cimgui.VC.db` slipped in accidentally? :)

**Comment by neslib** [comment]
Oops. You can ignore that one.
Sorry about that. Should add that one to my global gitignore...

**Comment by procedural** [comment]
You forgot `struct` keyword for `drawData` here: https://github.com/Extrawurst/cimgui/pull/18/files#diff-9dac42b77ccda07e6c9f40fb19c3c046R432
And `struct ImGuiTextFilter` isn't declared anywhere it seems, I get compiler warnings...

File: cimgui/cimgui.h [hunk 1]
```diff
@@ -417,13 +417,19 @@ CIMGUI_API struct ImFont*   ImFontAtlas_AddFontFromMemoryCompressedTTF(struct Im
...
+CIMGUI_API CONST ImWchar*   ImFontAtlas_GetGlyphRangesChinese(struct ImFontAtlas* atlas);
+CIMGUI_API CONST ImWchar*   ImFontAtlas_GetGlyphRangesCyrillic(struct ImFontAtlas* atlas);
 
 CIMGUI_API void             ImGuiIO_AddInputCharacter(unsigned short c);
 CIMGUI_API void             ImGuiIO_AddInputCharactersUTF8(CONST char* utf8_chars);
 CIMGUI_API void             ImGuiIO_ClearInputCharacters();
 
 //ImDrawData
 CIMGUI_API void                 ImDrawData_DeIndexAllBuffers(struct ImDrawData* drawData);
+CIMGUI_API void					ImDrawData_ScaleClipRects(ImDrawData* drawData, struct ImVec2 sc);
```
Comment (procedural) [inline cimgui/cimgui.h]: `ImDrawData* drawData` should be `struct ImDrawData* drawData`

**Comment by neslib** [comment]
Sorry again. I only use the Visual Studio solution and it compiled fine with my changes.
I added a new commit that fixes these issues (hopefully ;) I also removed the VC.db file.

**Comment by Extrawurst** [comment]
Thank you guys so much. did not look into it in detail yet, @procedural does it look good to you now ?

**Comment by procedural** [comment]
@Extrawurst yep, it's good now

**Comment by Extrawurst** [comment]
@neslib is this also updating the API to a recent version or is this adding new bindings in the status om imgui that cimgui is binding to (v 1.49) ?

**Comment by neslib** [comment]
@Extrawurst these new bindings still apply to v 1.49. They compile against the imgui submodule included with cimgui.

**Comment by Extrawurst** [comment]
ok great, thanks

**Comment by Extrawurst** [comment]
thank you very much!

**Pull request merged**